temporal-sdk/src/main/java/io/temporal/internal/sync/PotentialDeadlockException.java

```diff
@@ -19,7 +19,7 @@
 
 package io.temporal.internal.sync;
 
-class PotentialDeadlockException extends RuntimeException {
+public class PotentialDeadlockException extends RuntimeException {
```

Comment:
Should we add a javadoc for the class since it's public now?

Reply:
Changing it to public is pretty technical, it still stays in internal package, but I will write a javadoc here too, because this exception is something that users can encounter.